### PR TITLE
chore: Fix typing in ctl backup and restore

### PR DESCRIPTION
**Pull request by sharat87**
Now that we've moved to TypeScript, this PR fixes missing type annotations in `backup.ts` and `restore.ts`.

## Automation

/test sanity

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/11988138792>
> Commit: 5770dd0c1fe2c97787a6d8d454b36005d2f169f3
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=11988138792&attempt=2" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.Sanity`
> Spec:
> <hr>Sun, 24 Nov 2024 01:02:33 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **New Features**
	- Enhanced error handling for password mismatches during backup encryption.
	- Improved test coverage for backup functionality, including backup paths and disk space validation.

- **Bug Fixes**
	- Refined backup cleanup logic to ensure proper file retention and removal.

- **Refactor**
	- Introduced a new `BackupState` class to manage backup process state.
	- Streamlined password handling and improved code structure for better readability.
	- Updated type safety for several functions in the backup and restore processes.
	- Simplified exit process in error handling for restore functionality.

### CODE REVIEW

File: app/client/packages/rts/src/ctl/backup.ts [hunk 1]
```diff
@@ -339,13 +375,15 @@
   return new Date().toISOString().replace(/:/g, "-");
 }
 
-export async function getAvailableBackupSpaceInBytes(path) {
+export async function getAvailableBackupSpaceInBytes(
+  path: string,
+): Promise<number> {
```
[the remark this answers is not in the document]
Comment (sharat87) [reply app/client/packages/rts/src/ctl/backup.ts:380]: For later PR.

File: app/client/packages/rts/src/ctl/backup.ts [hunk 2]
```diff
@@ -106,27 +122,36 @@
...
+        await fsPromises.rm(state.archivePath, {
+          recursive: true,
+          force: true,
+        });
       }
     }
 
     await postBackupCleanup();
-    process.exit(errorCode);
+    process.exit();
```
[the remark this answers is not in the document]
Comment (sharat87) [reply app/client/packages/rts/src/ctl/backup.ts:147]: For later PR.

File: app/client/packages/rts/src/ctl/backup.ts [hunk 3]
```diff
@@ -10,26 +10,49 @@ import readlineSync from "readline-sync";
...
+
+  backupRootPath: string;
+  archivePath: string;
 
+  encryptionPassword: string;
+
+  isEncryptionEnabled() {
+    return !!this.encryptionPassword;
+  }
+}
```
[the remark this answers is not in the document]
Comment (sharat87) [reply app/client/packages/rts/src/ctl/backup.ts]: Done.

File: app/client/packages/rts/src/ctl/restore.ts [hunk 4]
```diff
@@ -307,18 +306,17 @@ async function getBackupDatabaseName(restoreContentsPath) {
...
+  let backupFilePath: string;
 
   await utils.ensureSupervisorIsRunning();
 
   try {
     let backupFileName = await getBackupFileName();
 
     if (backupFileName == null) {
-      process.exit(errorCode);
+      process.exit();
```
[the remark this answers is not in the document]
Comment (sharat87) [reply app/client/packages/rts/src/ctl/restore.ts:325]: For later PR.

File: app/client/packages/rts/src/ctl/backup.ts [hunk 5]
```diff
@@ -231,7 +261,10 @@ async function exportDockerEnvFile(destFolder, encryptArchive) {
   console.log("Exporting docker environment file done.");
 }
 
-export async function executeMongoDumpCMD(destFolder, appsmithMongoURI) {
+export async function executeMongoDumpCMD(
+  destFolder: string,
+  appsmithMongoURI: string,
+) {
```
[the remark this answers is not in the document]
Comment (sharat87) [reply app/client/packages/rts/src/ctl/backup.ts:267]: Not applicable.

**Review by mohanarpit — APPROVED** [review]

**Pull request merged**